cylc/flow/option_parsers.py

```diff
@@ -530,7 +532,7 @@ def get_cylc_rose_options():
                 help=(
                     "Use optional Rose Config Setting"
                     " (If Cylc-Rose is installed)"),
-                action="append", default=[], dest="opt_conf_keys",
+                action="append", default=None, dest="opt_conf_keys",
```

Comment:
These `default=[]` statements appear to have been passing a mutable object through a mechanism which mutated them?!

To prove this, try printing `option.default` after option parsing, you'll find it matches the value?!

Dunno, simple fix, don't pass in something mutable.